### ISSUE TITLE
[dove][package error] Creating an error with a null err breaks

### ISSUE THREAD

**Issue by mdartic** [feathersjs/feathers#2789]
I'm migrating the `feathers-authentication-management` package to dove, and I encounter an error in the... `error` package :-).

We have a call to `BadRequest` with potentially no error message in first param (see https://github.com/feathersjs-ecosystem/feathers-authentication-management/blob/master/src/methods/check-unique.ts#L74-L77).

In [crow implementation](https://github.com/feathersjs/feathers/blob/crow/packages/errors/lib/index.js), if we make the call

```ts
const err = new BadRequest(null, data)
// this will equivalent to
const err = FeathersError(null, 'BadRequest', 400, 'bad-request', data);
```
`err` will be like 
```js
{
  type: 'FeathersError',
  name: 'BadRequest',
  message: 'Error', // by default, with null value in first param, the "Error" string is set
  code: 400,
  className: 'bad-request', 
  ...
}
```

In [dove implementation](https://github.com/feathersjs/feathers/blob/dove/packages/errors/src/index.ts#L25), with the same call - the `BadRequest` one,
this will break at [#L35](https://github.com/feathersjs/feathers/blob/dove/packages/errors/src/index.ts#L35) as `typeof null === 'object'` is `true`.
Code will try to destructure `null` in `{ message, errors, ...rest }`.

Another interesting thing is that constructor of errors like, eg, [`BadRequest`](https://github.com/feathersjs/feathers/blob/dove/packages/errors/src/index.ts#L68) takes 2 optional params `message?: ErrorMessage, data?: any`, but the [`FeathersError` constructor](https://github.com/feathersjs/feathers/blob/dove/packages/errors/src/index.ts#L24) want mandatory params : `err` and `_data` aren't optional.
This use case allow `feathers-authentication-management` to still create `BadRequest` errors with an `err` `null`.

Is this use case still accepted ?
Could we fix the previous implementation ?
I fix temporary the package by updating the [#L36](https://github.com/feathersjs/feathers/blob/dove/packages/errors/src/index.ts#L36) : 

```js
  const { message, errors, ...rest } = typeof err === 'object' && err !== null ? err : _data;
```

If needed, I can submit a PR with fix on typing (FeathersError constructor with `err: ErrorMessage | null`) and tests proving `new BadRequest(null, ...)` work.

**Comment by daffl** [comment]
Thanks for the detailed error report. This should be fixed via #2776

**Comment by mdartic** [comment]
Thanks !
